src/main/java/apoc/cypher/Cypher.java

```diff
@@ -436,4 +445,39 @@ private static Collection asCollection(Object value) {
     public Stream<MapResult> doWhenCase(@Name("conditionals") List<Object> conditionals, @Name(value="elseQuery", defaultValue = "") String elseQuery, @Name(value="params", defaultValue = "") Map<String, Object> params) {
         return whenCase(conditionals, elseQuery, params);
     }
+
```

Comment:
shouldn't this go somewhere else?
I think I have similar code already for periodic.iterate?

Also I'm not sure that 

1. you get an error from the pooled execution as it usually just kills the thread in the execution, most likely at Future.get and then you get an InvocationTargetException where you have unpack the nested exception
1. it is a transient error? Not many errors make sense to retry, mostly deadlocks only or if some pre-codition is not met yet.

Reply:
Yes the code was taken from periodic.iterate and was repurposed for `apoc.cypher` calls. The periodic.iterate method I believe you are referring to is`iterateAndExecuteBatchedInSeparateThread`. It has parameters, such as batching and parallel processing which do not make sense in this context. This PR looks to allow a configurable retry. 

I can move this `runWithRetry` method to a Utils class if you think it would be better there.

1. Also I didn’t think the exception itself mattered so I didn't think it needed to be unpacked. It just matters that there is an exception and it is caught so a new transaction can be retried. 

2. Though there may be multiple uses, this PR has one specific use I was thinking of. This is useful for generating unique values in a trigger and then retrying if the value is already in use. 
For example, suppose we have a trigger set up called “addUniqueValueToCreatedNodes” that generates a `uuid` that must be unique (has a uniqueness CONSTRAINT on it) and adds it to created nodes. If a node is added to the graph the trigger will start and generate the `uuid` and try to add it.  If the generated `uuid` is already in use, the trigger will fail and the node will not be created. The ability to run the generation of `uuid` and being able to SET it on the node in `apoc.cypher.doIt` will allow for multiple retries in the trigger itself, so the node will be created. 

Here is the example in code:

`CALL apoc.trigger.add("addUniqueValueToCreatedNodes", "UNWIND {createdNodes} AS n CALL apoc.cypher.doIt(\"WITH {node} AS node SET node.uuid = 'uuid:' + apoc.create.uuid()\", {node: n}, {retries: 5}) YIELD value RETURN value", {phase: "after”})`

This would fail after 5 retries, making it much more likely the node would be created compared to the code below where there is no retry and a conflicting `uuid` would fail to create the node.

`CALL apoc.trigger.add("addUniqueValueToCreatedNodes", "UNWIND {createdNodes} AS n WITH n SET n.uuid = 'uuid:' + apoc.create.uuid()", {phase: "before”})`

Reply:
Ok, makes sense.

We meant to add an uuid adding feature to apoc anyway though :)

Btw. it's not recommended to uses prefixes like `uuid:` as that affects the ability for string compression as it leaves the character set of hex-code+separator.